### ISSUE TITLE
Please help! I cannot for the life of me get push notifications to work in my app, with this or other plugins.

### ISSUE THREAD

**Issue by jamminjames** [fechanique/cordova-plugin-fcm#197]
I just can't get push notifications to work in my Phonegap Build app -- I know I'm missing something, but have tried to follow all the instructions, and I don't know what.

I've tried other plugins, and this one, combined with the cordova-plugin-fcm-config plugin (forking it, as recommended, so that the google-services.json is installed correctly in my app), seems to have gotten me the closest. However, the push notifications are still not coming through to the end user, whether via Firebase or the test site that cordova-plugin-fcm-config gives. 

I'm at my wits end here, having tried various other push notification plugins, signing up with Google Firebase, etc. Does the Phonegap Build process make this impossible? Or what?

Any help, hints, how-tos, etc, would be much appreciated!

**Comment by Firzenizer** [comment]
Do you have paid PhoneGap Build subscription? It's required to install 3rd party plugins such as the forked cordova-plugin-fcm-config plugin.

**Comment by jamminjames** [comment]
I don't think you need a sub if you have just one app. Besides, it throws no errors or warnings, which PGB does rather prolifically, so I don't think there's a problem there. The log shows everything getting compiled correctly, including that plugin.

**Comment by Firzenizer** [comment]
https://build.phonegap.com/
Oh jeah. It says on the front page:
third party plugins (FREE PLAN CAN USE THESE)
* includes plugins from npmjs.com as well as our own repository
upload plugins (Non-npm plugins requires paid plan)
* includes plugins only you can use

Maybe there is issues with your code used to receive the notifications?

Does your FCMPlugin.getToken() return the token?

**Comment by jamminjames** [comment]
Firzenizer, thanks for your help. I'm a bit of a newbie, so I hope you'll be patient and stick with me!

I really think the problem is not Phonegap, as I'm getting no errors, and the log indicates everything is installed properly.

How do you activate the FCMPlugin.getToken() function? I tried using a button, like so:

`<p id="push-test"><a href="" data-role="button" data-icon="star">Push Test</a></p>`

`$(function(){
    $('#push-test').click(function() {
//	alert('Push test');
	FCMPlugin.getToken(
          function(token){
          alert(token);
          },
          function(err){
          alert('error retrieving token: ' + err);
          }
        )
    });
});`

The initial alert, when not commented out, indicates the button itself is working. However, I'm not getting any alert from the FCMPlugin.getToken() section, success or error.

**Comment by Firzenizer** [comment]
That's weird. It should trigger either success or error no matter what. Are you sure FCMPlugin is defined and you aren't getting any JS errors?

Maybe try 

```
function onDeviceReady() {
    if(typeof FCMPlugin !== "undefined"){
         alert("FCMPlugin is installed correctly");
    }
}

document.addEventListener("deviceready", onDeviceReady, false);
```

**Comment by jamminjames** [comment]
Thanks. Did that, and it popped up.

**Comment by jamminjames** [comment]
What should I do now?

**Comment by jamminjames** [comment]
So, looking at the Phonegap Build logs, I found this:
cp: dest file already exists: /project/cordova/plugins/cordova-plugin-fcm-config/GoogleService-Info.plist
cp: dest file already exists: /project/cordova/plugins/cordova-plugin-fcm-config/google-services.json
cp: dest file already exists: /project/cordova/plugins/cordova-plugin-fcm-config/package.json
cp: dest file already exists: /project/cordova/plugins/cordova-plugin-fcm-config/plugin.xml

So, I guess it's not working to install it this way, like I thought. The main plugin, cordova-plugin-fcm, is installing just fine. I guess the main obstacle is installing my own google-services.json file, since I'm only doing Android at this point. I was hoping this plugin would do that. 

Any ideas on how I can do this in Phonegap Build? I tried just putting google-services.json in my root folder, but that doesn't seem to work.

**Comment by Firzenizer** [comment]
You should probably move the getToken code to onDeviceReady.

```
function onDeviceReady() {
    if(typeof FCMPlugin !== "undefined"){
         FCMPlugin.getToken(
              function(token){
                    alert(token);
              }, function(err){
                    alert('error retrieving token: ' + err);
              }
         );
    }
}

document.addEventListener("deviceready", onDeviceReady, false);
```

**Comment by imbabaer** [comment]
> function onDeviceReady() {
    if(typeof FCMPlugin !== "undefined"){
         alert("FCMPlugin is installed correctly");
    }
}
document.addEventListener("deviceready", onDeviceReady, false);


@Firzenizer this doesn't pop up in my case. I installed the plugin with `cordova plugin add cordova-plugin-fcm`, an error occurs `Error loading Cordova-Browser` but according to this [link ](http://stackoverflow.com/a/40922518/7272308) there is no problem with that.
Any suggestions?
